src/mvs/fusion.h

```diff
@@ -120,41 +137,36 @@ class StereoFusion : public Thread {
   std::vector<char> used_images_;
   std::vector<char> fused_images_;
   std::vector<std::vector<int>> overlapping_images_;
-  std::vector<Mat<bool>> fused_pixel_masks_;
+  // Contains image masks of pre-masked and already fused pixels.
+  // Initialized from image masks if provided in StereoFusionOptions.
+  std::vector<Mat<char>> fused_pixel_masks_;
   std::vector<std::pair<int, int>> depth_map_sizes_;
   std::vector<std::pair<float, float>> bitmap_scales_;
   std::vector<Eigen::Matrix<float, 3, 4, Eigen::RowMajor>> P_;
   std::vector<Eigen::Matrix<float, 3, 4, Eigen::RowMajor>> inv_P_;
   std::vector<Eigen::Matrix<float, 3, 3, Eigen::RowMajor>> inv_R_;
 
   struct FusionData {
-    int image_idx = kInvalidImageId;
-    int row = 0;
-    int col = 0;
-    int traversal_depth = -1;
+    int image_idx;
+    int row;
+    int col;
+    int traversal_depth;
```

Comment:
Can you restore the previous default values here? It's generally good practice to keep these to avoid uninitialized memory, I think.

Reply:
Sounds good; I only removed them because now we never create a default `FusionData` struct and we only use it with the constructor through `emplace_back`